File: lib/file/normalizeFile.ts
// NFD - macOS
// NFC - Windows
export function normalizeFile(
	filePath: string,
	normalizationForm?: 'NFD' | 'NFC' | 'NFKC' | 'NFKD' | undefined
): string {
	if (!filePath || filePath.length < 1) {
		return '';
	}

	return filePath.normalize(normalizationForm);
}

File: packages/python/qsu/file/normalizeFile.py
import unicodedata
from typing import Literal, Optional, cast

def normalizeFile(filePath: str, normalizationForm: Optional[str] = None) -> str:
	# NFD - macOS
	# NFC - Windows
	if not filePath or len(filePath) < 1:
		return ''

	# JS String.prototype.normalize() defaults to 'NFC'.
	form = normalizationForm if normalizationForm else 'NFC'

	return unicodedata.normalize(cast(Literal['NFC', 'NFD', 'NFKC', 'NFKD'], form), filePath)
